cvat/apps/engine/mixins.py

```diff
@@ -40,50 +98,42 @@ def write_chunk(self, chunk):
         with open(self.file_path, 'r+b') as file:
             file.seek(chunk.offset)
             file.write(chunk.content)
-        self.offset = cache.incr("tus-uploads/{}/offset".format(self.file_id), chunk.size)
+        self.meta_file.meta.offset += chunk.size
+        self.meta_file.save()
 
     def is_complete(self):
         return self.offset == self.file_size
 
     def rename(self):
-        file_id_path = os.path.join(self.upload_dir, self.file_id)
         file_path = os.path.join(self.upload_dir, self.filename)
-        file_exists = os.path.lexists(os.path.join(self.upload_dir, self.filename))
-        if file_exists:
+        if os.path.lexists(file_path):
             original_file_name, extension = os.path.splitext(self.filename)
             file_amount = 1
             while os.path.lexists(os.path.join(self.upload_dir, self.filename)):
-                self.filename = "{}_{}{}".format(original_file_name, file_amount, extension)
+                self.meta_file.meta.filename = "{}_{}{}".format(original_file_name, file_amount, extension)
                 file_path = os.path.join(self.upload_dir, self.filename)
                 file_amount += 1
-        os.rename(file_id_path, file_path)
+        os.rename(self.file_path, file_path)
 
     def clean(self):
-        cache.delete_many([
-            "tus-uploads/{}/file_size".format(self.file_id),
-            "tus-uploads/{}/filename".format(self.file_id),
-            "tus-uploads/{}/offset".format(self.file_id),
-            "tus-uploads/{}/metadata".format(self.file_id),
-        ])
-
-    @staticmethod
-    def get_tusfile(file_id, upload_dir):
-        file_exists = cache.get("tus-uploads/{}/filename".format(file_id), None) is not None
-        if file_exists:
-            return TusFile(file_id, upload_dir)
-        return None
+        self.meta_file.delete()
 
     @staticmethod
     def create_file(metadata, file_size, upload_dir):
         file_id = str(uuid.uuid4())
         filename = metadata.get("filename")
-        cache.add("tus-uploads/{}/filename".format(file_id), "{}".format(filename), TusFile._tus_cache_timeout)
-        cache.add("tus-uploads/{}/file_size".format(file_id), file_size, TusFile._tus_cache_timeout)
-        cache.add("tus-uploads/{}/offset".format(file_id), 0, TusFile._tus_cache_timeout)
-        cache.add("tus-uploads/{}/metadata".format(file_id), metadata, TusFile._tus_cache_timeout)
```

Comment:
Tus supports custom metadata which can be provided in HTTP_UPLOAD_METADATA header, its supported(but not used in UI). Here we saved it in cache.
Now as I see it wont be written in TusMeta file, do you think we should support each field individually?

Reply:
I haven't seen the `metadata' from the cache used anywhere in the server code, so I removed it,
but we can keep it inside a file for the future

Reply:
I think it would be good for the sake of supporting tus protocol